### PR TITLE
Make sendgird variables customizable

### PR DESCRIPTION
**Pull request by makinoshi**
SendGrid API URLs are hard coded but there are some cases that we would like to customize them for example under development with SendGrid mock service.

This PR make it enable to inject SendGrid base URL, which is `https://api.sendgrid.com` in production.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
 # Change Log
 
 ## [Unreleased]
+### Breaking
```
Comment (liquidz) [inline CHANGELOG.md:6]: Is this change breaking?
`:api-base-url` is optional and when `:api-base-url` is not specified, `send-url` and `bounce-url` are same as before.
Of course, `email*` , `retrieve-all-bounces*` and `delete-bounces*` functions are breaking, but these functions are private.
So it should be `Changed` in my opinion.

**Review by liquidz — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (makinoshi) [reply CHANGELOG.md:6]: `send-url` and `bounce-url` were public var before so I wrote as breaking.
But they were less likely used, I can agree with your opinion.

What do you think of?

Comment (liquidz) [reply CHANGELOG.md:6]: Oh, I overlooked `send-url` and `bounce-url` were public.
It is breaking indeed.
I reconsidered that `Breaking` is fine as it is.

**Review by liquidz — APPROVED** [review]

**Comment by makinoshi** [comment]
Merge main into this branch because changelog was conflicted

**Pull request merged**